### PR TITLE
{data,math,tools}[GCCcore/13.3.0,foss/2024a,gompi/2024a] gperftools v2.16, HDF5 v1.14.5, HPX v1.10.0

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

- Depends on Szip from PR #21510

### CODE REVIEW

**Comment by smoors** [comment]
@maxim-masterov i moved `github_account` back to where it belongs, and also changed the sanity_check paths to the original one (using `SHLIB_EXT`)

please don't change the formatting/structuring just for the sake of it. in most cases there's a good reason why it's done this way.

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen3

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

File: easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ('HDF5', '1.14.5'),
+    ('Boost', '1.85.0'),
+    ('hwloc', '2.10.0'),
+    ('gperftools', '2.16'),
+]
+
+build_type = 'Release'
+
+configopts = '-DCMAKE_CXX_COMPILER=g++ '
+configopts += '-DCMAKE_BUILD_TYPE=Release '
```
Comment (smoors) [inline easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb:37]: don't use CMAKE_BUILD_TYPE=Release
```suggestion
```

File: easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+]
+dependencies = [
+    ('HDF5', '1.14.5'),
+    ('Boost', '1.85.0'),
+    ('hwloc', '2.10.0'),
+    ('gperftools', '2.16'),
+]
+
+build_type = 'Release'
+
```
Comment (smoors) [inline easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb:35]: remove: it's already the default, and the test complains about it.
```suggestion
```

File: easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+configopts += '-DHPX_WITH_NETWORKING=TRUE '
+configopts += '-DHPX_WITH_PARCELPORT_TCP=FALSE '
+configopts += '-DHPX_WITH_PARCELPORT_MPI=TRUE '
+configopts += '-DHPX_WITH_TESTS=FALSE '
+configopts += '-DHPX_WITH_EXAMPLES=TRUE '
+# configopts += '-DHPX_WITH_MAX_CPU_COUNT=128' #this should be handled by a hook for the system
+# configopts += '-DHPX_WITH_MAX_CPU_COUNT=' + os.cpu_count()
+configopts += '-DHPX_WITH_FETCH_ASIO=TRUE '
+
+bin_lib_subdirs = ['lib/%(namelower)s/']
```
Comment (smoors) [inline easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb:49]: remove unused param
```suggestion
bin_lib_subdirs = ['lib/%(namelower)s/']
```

File: easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['bin', 'lib'],
```
Comment (smoors) [inline easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb:53]: can you be a bit more specific in the files that are installed

**Comment by maxim-masterov** [comment]
> @maxim-masterov i moved `github_account` back to where it belongs, and also changed the sanity_check paths to the original one (using `SHLIB_EXT`)
> 
> please don't change the formatting/structuring just for the sake of it. in most cases there's a good reason why it's done this way.

Thanks, I missed that part. I guess it was automatically reformatted by the `eb` command when I used `--try-update-deps` and `--copy-ec` options. I noted it for the later PRs.

File: easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb (hunk 3 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/h/HPX/HPX-1.10.0-foss-2024a.eb:47]: `bin_lib_subdirs` is needed to make sure that relevant path is added to `LD_LIBRARY_PATH` before the sanity check

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
skl-rockylinux-810 - Linux Rocky Linux 8.10, x86_64, Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz (skylake), 1 x NVIDIA NVIDIA RTX A4000, 555.42.06, Python 3.6.8
See https://gist.github.com/SebastianAchilles/60392c34707b74121529019cb8214e0b for a full test report.

**Comment by SebastianAchilles** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**